### PR TITLE
adapt to latest ProxyManager

### LINKED ISSUE #3157: Sulu cache documents folder not created
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | dev-develop (25.01.2017)

#### Actual Behavior

On my local system on prod the `tmp_local/website/prod/sulu/documents` is not created automatically. Maybe a problem because of changing the folder to `tmp_local` but cant reproduce it on other system. All other cache folders seem to be created correctly. Only this one is missing.

#### Expected Behavior

The folder should be created correctly.

#### Steps to Reproduce

AbstractKernel.php

```php
    public function getCacheDir()
    {
        if ($this->environment == 'prod') {
            return dirname($this->rootDir) . DIRECTORY_SEPARATOR
                   . 'tmp_local' . DIRECTORY_SEPARATOR
                   . 'cache' . DIRECTORY_SEPARATOR
                   . $this->getContext() . DIRECTORY_SEPARATOR
                   . $this->environment;
        }

        return dirname($this->rootDir) . DIRECTORY_SEPARATOR
               . 'var' . DIRECTORY_SEPARATOR
               . 'cache' . DIRECTORY_SEPARATOR
               . $this->getContext() . DIRECTORY_SEPARATOR
               . $this->environment;
    }
```

 - rm -rf tmp_local/cache/*
 - http://sulu.lo/any-url

Stack Trace:

```php
Fatal error: Uncaught ProxyManager\Exception\InvalidProxyDirectoryException: Provided directory 
"/Users/alexanderschranz/Development/Sulu/sulu.lo/tmp_local/cache/website/prod/sulu/documents" does not exist in 
/Users/alexanderschranz/Development/Sulu/sulu.lo/vendor/ocramius/proxy-manager/src/ProxyManager/Exception/InvalidProxyDirectoryException.php:38 Stack trace:
#0 /Users/alexanderschranz/Development/Sulu/sulu.lo/vendor/ocramius/proxy-manager/src/ProxyManager/FileLocator/FileLocator.php(46): ProxyManager\Exception\InvalidProxyDirectoryException::proxyDirectoryNotFound('/Users/alexande...')
#1 /Users/alexanderschranz/Development/Sulu/sulu.lo/vendor/ocramius/proxy-manager/src/ProxyManager/Configuration.php(171): ProxyManager\FileLocator\FileLocator->__construct('/Users/alexande...')
#2 /Users/alexanderschranz/Development/Sulu/sulu.lo/vendor/ocramius/proxy-manager/src/ProxyManager/Factory/AbstractBaseFactory.php(109): ProxyManager\Configuration->getGeneratorStrategy()
#3 /Us in /Users/alexanderschranz/Development/Sulu/sulu.lo/src/AppBundle/Resources/views/blocks/header.html.twig on line 51
```
Line 51:
```twig
{% for item in sulu_navigation_root_tree('main', 2, true) if item.uuid is defined %}
```

#### Possible Solutions

Create the documents folder as early as possible.

### PR DESCRIPTION
**Pull request by danrot**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #3157 
| Related issues/PRs | #3347, depends on https://github.com/sulu/sulu-document-manager/pull/113
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR upgrades the `ocramius/proxy-manager` package, because the currently installed version is not compatible with PHP7.

#### Why?

There was an error when accessing a property from a resource given by the SmartContent.

### CODE REVIEW

**Comment by chirimoya** [comment]
@danrot ... move the proxy manager configuration and strategy to the core and use it in the different places.

File: composer.json [hunk 1]
```diff
@@ -105,5 +105,11 @@
...
             "Sulu\\": "tests/"
         }
-    }
+    },
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "git@github.com:danrot/sulu-document-manager.git"
+        }
+    ]
```
Comment (danrot) [inline composer.json]: Should be removed before merge

File: UPGRADE.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 ## dev-develop
 
+### ProxyManager
+
+We had to update `ocramius/proxy-manager` in order to be compatible with PHP 7.
+In case you have defined your own proxies, you should checkthe
```
Comment (yukoff) [inline UPGRADE.md]: Sorry for getting in-beetween - small typo here: "checkthe" -> "check the" ;)

Comment (danrot) [inline UPGRADE.md]: No need to be to sry, thanks a lot 😃

**Pull request merged**